ml-agents/mlagents/trainers/demo_loader.py

```diff
@@ -93,20 +93,28 @@ def demo_to_buffer(
     demo_buffer = make_demo_buffer(info_action_pair, behavior_spec, sequence_length)
     if expected_behavior_spec:
         # check action dimensions in demonstration match
-        if behavior_spec.action_shape != expected_behavior_spec.action_shape:
+        if (
```

Comment:
Move this logic to a method on ActionSpec too? E.g. `def action_space_equal(other_spec: ActionSpec) -> bool`

Reply:
I think we can have implementations of `__eq__(self, other) ` on all "specs" objects.

Reply:
Yeah, guess I wasn't sure we needed exact equality in the action specs here or not.

Reply:
(but agreed the logic should be on the spec objects)

Reply:
Done